src/globus_sdk/exc/api.py

```diff
@@ -0,0 +1,167 @@
+        """
+        Get the verbatim error message received from a Globus API, interpreted
+        as JSON data
+
+        If the body cannot be loaded as JSON, this is None
+        """
+        r = self._underlying_response
+        if "Content-Type" in r.headers and (
+            "application/json" in r.headers["Content-Type"]
+        ):
+            try:
+                return r.json()
+            except ValueError:
+                log.error(
+                    "Error body could not be JSON decoded! "
+                    "This means the Content-Type is wrong, or the "
+                    "body is malformed!"
+                )
+                return None
+        else:
+            return None
+
+    @property
+    def raw_text(self):
+        """
+        Get the verbatim error message receved from a Globus API as a *string*
+        """
+        return self._underlying_response.text
+
+    @property
+    def info(self) -> ErrorInfoContainer:
+        """
+        An ``ErrorInfoContainer`` with parsed error data. The ``info`` of an error is
+        guaranteed to be present, but all of its contents may be falsey if the error
+        could not be parsed.
+        """
+        if self._info is None:
+            rawjson = self.raw_json
+            json_data = rawjson if isinstance(rawjson, dict) else None
+            self._info = ErrorInfoContainer(cast(Optional[Dict[str, Any]], json_data))
+        return self._info
+
+    def _get_request_authorization_scheme(self):
+        try:
+            authz_h = self._underlying_response.request.headers["Authorization"]
+            try:
+                authz_scheme = authz_h.split()[0]
+                if authz_scheme.lower() in self.RECOGNIZED_AUTHZ_SCHEMES:
+                    return authz_scheme
+            except IndexError:
+                pass
+        except KeyError:
+            pass
+        return None
+
+    def _get_args(self):
+        """
+        Get arguments to pass to the Exception base class. These args are
+        displayed in stack traces.
+        """
+        return [
+            self._underlying_response.request.method,
+            self._underlying_response.url,
+            self._get_request_authorization_scheme(),
+            self.http_status,
+            self.code,
+            self.message,
+        ]
+
+    def _parse_response(self):
+        """
+        This is an intermediate step between 'raw_json' (loading bare JSON data)
+        and the "real" parsing method, '_load_from_json'
+
+        _parse_response() pulls the JSON body and does the following:
+        - if the data is not a dict, ensure _load_from_json is not called (so it only
+          gets called on dict data)
+        - if the data contains an 'errors' array, pull out the first error document and
+          pass that to _load_from_json()
+        - log a warning on non-dict JSON data
+        """
+        json_data = self.raw_json
+        if json_data is None:
```

Comment:
It looks like this could be organized to "de-dent" the main block here, which I'm a proponent of:
```python
if json_data is None:
    ...
    return
if not isinstance(json_data, dict):
    log.warning("could not be parsed as JSON")
    return
# de-dented `isinstance(json_data, dict)` block goes here
...
```

Reply:
Yep, I'm in favor. 👍